spec/compiler/semantic/instance_var_spec.cr

```diff
@@ -5387,4 +5387,166 @@ describe "Semantic: instance var" do
       ),
       "can't infer the type of instance variable '@x' of Foo"
   end
+
+  it "errors when overriding inherited instance variable with incompatible type" do
+    assert_error <<-CR, "instance variable '@a' of A must be Int32, not (Char | Int32)"
+      class A
+        @a = 1
+      end
+
+      class B < A
+        @a = 'a'
+      end
+      CR
+  end
+
+  it "accepts overriding inherited instance variable with compatible type" do
+    semantic <<-CR
+      class A
+        @a = 1
+      end
+
+      class B < A
+        @a = 2
+      end
+      CR
+  end
+
+  describe "instance variable inherited from multiple parents" do
+    context "with compatible type" do
+      it "module and class, with declarations" do
+        assert_error <<-CR, "instance variable '@a' of B is already defined in A"
+          module M
+            @a : Int32 = 1
+          end
+
+          class A
+            @a : Int32 = 2
+          end
+
+          class B < A
+            include M
+          end
+          CR
+      end
+
+      it "module and class, with definitions" do
```

Comment:
This looks like a duplicate of the one right before it (including the `it` description).  The only difference I see is the explicit typing of `@a`.  Was that an intentional difference, and if so, perhaps the `it` description can be changed to describe why?